dapps/evancore.vue.libs/src/dispatcher.ts

```diff
@@ -0,0 +1,245 @@
+   * dispatcher, will be loaded.
+   */
+  loadedDispatchers: Array<string> = [ ];
+
+  /**
+   * Current bcc runtime in user context
+   */
+  runtime: bcc.Runtime;
+
+  /**
+   * All dispatcher instances that are currently in error state, mapped from it's full ens address to a bool flag.
+   */
+  error: any = { };
+
+  /**
+   * All dispatcher instances that are currently running, mapped from it's full ens address to a bool flag.
+   */
+  running: any = { };
+
+  /**
+   * All dispatcher instances, mapped from it's full ens address to a array of dispatcher instances
+   */
+  instances: any = { };
+
+  /**
+   * All dispatcher instances, mapped from it's full ens address to a array of dispatcher payload objects
+   */
+  data: any = { };
+
+  /**
+   * Includes only dispatcher instances that are running for the active dapp, mapped to the error,
+   * running and data scope.
+   */
+  curr: any = {
+    data: { },
+    error: { },
+    instances: { },
+    running: { },
+  };
+
+  /**
+   * Initialize a new vue dispatcher handler
+   *
+   * @param      {<type>}  this    The object
+   */
+  constructor(vue: Vue) {
+    this.vue = vue;
+    this.runtime = vue.$store.state.runtime;
+    this.dapp = vue.$store.state.dapp;
+    this.dappName = this.dapp.ens;
+
+    vue.$store.state.dispatcher = this;
+  }
+
+  /**
+   * Load latest dispatcher information.
+   */
+  async initialize() {
+    const queue = await new EvanQueue(this.runtime.activeAccount);
+    const dispatchers = await queue.load('*');
+
+    // load all dispatcher instances for this user
+    await Promise.all(dispatchers.map(async (dispatcherObj: any) => {
+      try {
+        const dispatcher = await this.loadDispatcher(dispatcherObj.dispatcherId);
+        await Promise.all(Object.keys(dispatcherObj.entries).map(async (instanceId: string) => {
+          const entry = dispatcherObj.entries[instanceId];
+          const instance = new DispatcherInstance({
+            queue,
+            dispatcher,
+            runtime: this.runtime,
+            data: entry.data,
+            stepIndex: entry.stepIndex,
+            id: instanceId,
+            error: entry.error,
+            customPrice: entry.customPrice,
+          });
+
+          this.handleDispatcherEvent(dispatcher, instance);
+        }));
+      } catch (ex) {
+        console.error(ex.message);
+      }
+    }));
+
+    // start dispatcher watching
+    this.dispatcherWatcher = Dispatcher.watch(
+      ($event: any) => this.handleDispatcherEvent($event.detail.dispatcher, $event.detail.instance),
+    );
+  }
+
+  /**
+   * Remove the dispatcher watcher.
+   */
+  destroy() {
+    this.dispatcherWatcher && this.dispatcherWatcher();
+  }
+
+  /**
+   * Load dispatcher from an ens address or return thee cached one.
+   */
+  async loadDispatcher(dispatcherId: string) {
+    const [ dappEns, dispatcherName ] = dispatcherId.split('|||');
```

Comment:
||| 👀

Reply:
used for using dispatcher ens address and id's as single keys in index db.